### ISSUE TITLE
HelloSign-sdk is broken when upgrading to nock 13.x

### ISSUE THREAD

**Issue by toymachiner62** [HelloFax/hellosign-nodejs-sdk#80]
I just upgraded my project from nock 12.0.3 to nock 13.0.4 and my tests break. 

Here's an example project using nock 12.0.3 and I can easily stub out the response from the hellosign-sdk client https://repl.it/@TomCaflisch/HelplessWorthlessCell#index.js

Here's an example project using nock 13.0.4 and you can see the hellosign-sdk error
https://repl.it/@TomCaflisch/UnrealisticPopularGlueware#index.js

```
err =  Error: write after end
    at Constructor._Error (/home/runner/UnrealisticPopularGlueware/node_modules/hellosign-sdk/lib/Error.js:36:19)
    at new Constructor (/home/runner/UnrealisticPopularGlueware/node_modules/hellosign-sdk/lib/utils.js:112:19)
    at OverriddenClientRequest.<anonymous> (/home/runner/UnrealisticPopularGlueware/node_modules/hellosign-sdk/lib/HelloSignResource.js:213:17)
    at OverriddenClientRequest.emit (events.js:315:20)
    at OverriddenClientRequest.EventEmitter.emit (domain.js:483:12)
    at /home/runner/UnrealisticPopularGlueware/node_modules/nock/lib/intercepted_request_router.js:145:34
    at processTicksAndRejections (internal/process/task_queues.js:79:11) {
  type: 'HelloSignError',
  rawType: undefined,
  code: 'ERR_STREAM_WRITE_AFTER_END',
  param: undefined,
  detail: undefined,
  raw: Error: write after end
      at InterceptedRequestRouter.handleWrite (/home/runner/UnrealisticPopularGlueware/node_modules/nock/lib/intercepted_request_router.js:143:19)
      at OverriddenClientRequest.req.write (/home/runner/UnrealisticPopularGlueware/node_modules/nock/lib/intercepted_request_router.js:90:35)
      at InterceptedRequestRouter.handleEnd (/home/runner/UnrealisticPopularGlueware/node_modules/nock/lib/intercepted_request_router.js:198:9)
      at OverriddenClientRequest.req.end (/home/runner/UnrealisticPopularGlueware/node_modules/nock/lib/intercepted_request_router.js:91:33)
      at OverriddenClientRequest.<anonymous> (/home/runner/UnrealisticPopularGlueware/node_modules/hellosign-sdk/lib/HelloSignResource.js:422:21)
      at OverriddenClientRequest.emit (events.js:315:20)
      at OverriddenClientRequest.EventEmitter.emit (domain.js:483:12)
      at InterceptedRequestRouter.startPlayback (/home/runner/UnrealisticPopularGlueware/node_modules/nock/lib/intercepted_request_router.js:302:11)
      at InterceptedRequestRouter.maybeStartPlayback (/home/runner/UnrealisticPopularGlueware/node_modules/nock/lib/intercepted_request_router.js:253:12)
      at InterceptedRequestRouter.handleEnd (/home/runner/UnrealisticPopularGlueware/node_modules/nock/lib/intercepted_request_router.js:200:10) {
    code: 'ERR_STREAM_WRITE_AFTER_END'
  }
}
res =  null
```

I'm assuming the issue is with something that hellosign-sdk is doing since the other hundreds of tests I have are still working after upgrading to nock 13.

**Comment by toymachiner62** [comment]
Similar to #79

**Comment by latoyazamill** [comment]
Hi @toymachiner62 ,

This is LaToya from API support.

Would you mind getting back to us by creating a ticket to apisupport@hellosign.com with what the HelloSign SDK is doing in response when hitting the endpoint in your test suite outside of the Nock implementation?

For background: Our library is just making requests using the default https module in node; It seems like this is possibly an issue with Nock or the way that it is being used in this case. At first glance this seems to be an error being thrown from Nock versus HelloSign. Meaning, the same request works fine without errors in Node but will throw the error when using Nock. Specifically, this started happening in Nock v13 (migration guide). 

However, we are in need of more information from you and your exact use case. 

We will be on the look-out for your ticket in our queue and will be closing this issue. If the root of the issue is SDK focused, we will reopen and troubleshoot.

**Comment by toymachiner62** [comment]
> ... with what the HelloSign SDK is doing in response when hitting the endpoint in your test suite outside of the Nock implementation

This doesn't make sense. The issue is only when using (or rather upgrading from Nock 12.x - 13.x). I included an example that shows the issue in a very simple example.

I also created a ticket by sending an email as you suggested.

**Comment by hazemhamedhs** [comment]
Thank you for opening a ticket @toymachiner62. I will be in touch with you via the zendesk ticket #11587856‬ to continue communicating about this issue.

**Comment by toymachiner62** [comment]
For any google searches that land here, the issue was fixed in nock 13.0.5. https://github.com/nock/nock/issues/2112
